### PR TITLE
Feature/redux toolkit

### PR DESCRIPTION
**Pull request by JaHajRF**
redux/toolkit added; 
TrackList fetch moved to audius saga

### CODE REVIEW

File: packages/webapp/src/store/sagas/player.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      progress: 0,
+    })
+  );
+}
+
+export function* playerSaga(): Generator {
+  yield all([
+    takeEvery(
+      Player.HANDLE_CURRENT_TRACK,
+      (payload: PlayerActions["handleCurrentTrack"]) =>
```
Comment (kowalski) [inline packages/webapp/src/store/sagas/player.ts:42]: Don't unpack payload to positional arguments. Get payload in saga.

File: packages/webapp/src/store/sagas/audius.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  } catch (error) {
+    yield put(audiusEndpointActions.fetchFail(error));
+  }
+}
+
+export function* audiusSaga(): Generator {
+  yield all([
+    takeEvery(Audius.FETCH_ENDPOINTS, fetchEndpoints),
+    takeEvery(Audius.FETCH_TRACK_LIST, (payload: AudiusTrackActions["fetch"]) =>
```
Comment (kowalski) [inline packages/webapp/src/store/sagas/audius.ts:67]: don't unpack payload, pass it to saga function directly

File: packages/webapp/src/store/sagas/audius.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      throw new Error("Could not get endpoints");
+    }
+  } catch (error) {
+    yield put(audiusEndpointActions.fetchFail(error));
+  }
+}
+
+export function* fetchTrackList(searchInput: string) {
+  const endpoint = yield select(audiusSelectors.endpoint);
```
Comment (kowalski) [inline packages/webapp/src/store/sagas/audius.ts:32]: const endpoint = yield * getEndpoint();

instead `* getEndpoint()` handle situation when `select(..)` returns null (by fetching endpoint)

File: packages/webapp/src/store/sagas/audius.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  audiusTrackActions,
+  AudiusTrackActions,
+} from "../reducers/audius";
+import { Track } from "../reducers/types";
+import { audiusSelectors } from "../selectors/audius";
+
+export function* fetchEndpoints() {
+  try {
+    const response = (yield call(
+      axios.get,
```
Comment (kowalski) [inline packages/webapp/src/store/sagas/audius.ts:15]: consider switching to `fetch`. 

Can be in future PR

Comment (JaHajRF) [reply packages/webapp/src/store/sagas/audius.ts:15]: https://github.com/turntablefm/turntable/issues/11

**Review by kowalski — APPROVED** [review]

**Pull request merged**